pkg/secret/secret.go

```diff
@@ -0,0 +1,28 @@
+// Package contains interfaces for API secrets, storage and encoder.
+package secret
+
+//Provider defines what we can do with the data in our application
+//Describes the behavior of commands from the terminal
+type Provider interface {
+	// set by key and put value
```

Comment:
Use [go documentation](https://blog.golang.org/godoc) documenting convention to write documentation for methods/interfaces. `[Name of the method/interface] ...`